File: server/src/Models/DMToolkitModels/Instances/LineageInstance.cs
using DMToolkit.Models.Definitions;

namespace DMToolkit.Models.Instances;

public class LineageInstance : IDefinitionInstance<LineageDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public ICollection<FeatInstance> FeatInstances { get; set; } = new List<FeatInstance>();
    public string SpeciesInstanceId { get; set; } = string.Empty;
    public SpeciesInstance? SpeciesInstance { get; set; } = null; 
    public string DefinitionId { get; set; } = string.Empty;
    public LineageDefinition? Definition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/lineageInstance.ts
import type { LineageDefinition } from "../definitions/lineageDefinition";
import type { FeatInstance, FeatInstanceDto } from "./featInstance";

export interface LineageInstance {
    id: string;
    featInstances: FeatInstance[];
    definition: LineageDefinition;
}

export interface LineageInstanceDto {
    id: string;
    featInstances: FeatInstanceDto[];
    definitionId: string;
}